### PR TITLE
feat: rebalance

### PR DESCRIPTION
**Pull request by antoncoding**
## TLDR
* Add rebalance modal + design
* rebalance hook
* todo: (approve) + permit

## Release Notes

- **New Features**
  - Introduced a new component for managing and displaying two sets of market data, allowing users to filter and paginate through existing positions and available markets.
  - Added a modal for rebalancing financial positions, enabling users to input amounts, select markets, and view pending actions.
  - Enhanced pagination component with an optional settings button.
  - Added a new `MarketBadge` component to display market information visually.
  - Introduced a `RebalanceCart` component to summarize rebalance actions.

- **Improvements**
  - Updated accessibility rules for better compliance.
  - Enhanced data retrieval in user positions with additional fields for more detailed market information.
  - Expanded data structures for managing grouped market positions and transactions.
  - Added a new modal to guide users through the token rebalance process with clear steps.
  - Introduced a new `Actions` column in the positions summary table for improved user interaction.

- **Bug Fixes**
  - Adjusted the rendering logic for the positions summary table to include new action features.

### CODE REVIEW

File: src/hooks/useRebalance.ts [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+          },
+          false,
+        ],
+      });
+
+      transactions.push(authorizationTx);
+
+      // Wait for 0.5 seconds
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+    }
```
[the remark this answers is not in the document]
Comment (antoncoding) [reply src/hooks/useRebalance.ts:114]: this is needed because there's a known issue in Rabby wallet to not able to handle sequential requests
